File: Java/383. Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] cntMag = new int[26];
        int[] cntRan = new int[26];

        for (char c : magazine.toCharArray())
            cntMag[c - 'a']++;
        for (char c : ransomNote.toCharArray())
            cntRan[c - 'a']++;

        for (int i = 0; i < 26; i++)
            if (cntRan[i] > cntMag[i])
                return false;
        return true;
    }
}

File: Python/383. Ransom Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        cnt_mag = [0] * 26;
        cnt_ran = [0] * 26;

        for c in ransomNote:
            cnt_ran[ord(c) - ord('a')] +=1;
        for c in magazine:
            cnt_mag[ord(c) - ord('a')] += 1;
        
        for i in range(0 , 26):
            if cnt_ran[i] > cnt_mag[i]:
                return False
        return True
        